test/open_trip_planner_client/parser_test.exs

```diff
@@ -38,46 +38,85 @@ defmodule OpenTripPlannerClient.ParserTest do
+               },
+               %OpenTripPlannerClient.Error{
+                 details: %{
+                   extensions: %{classification: "ValidationError"},
+                   locations: [%{line: 1, column: 16}]
+                 },
+                 message: "Validation error (UnusedVariable) : Unused variable 'fromPlace'",
+                 type: :graphql_error
+               }
+             ]
+
       assert log =~ "Validation error"
     end
 
     test "handles GraphQL field error" do
-      assert {{:error, :graphql_field_error}, log} =
-               with_log(fn ->
-                 validate_body(%{
-                   data: %{plan: nil},
-                   errors: [
-                     %{
-                       message:
-                         "Exception while fetching data (/plan) : The value is not in range[0.0, 1.7976931348623157E308]: -5.0",
-                       locations: [
-                         %{
-                           line: 2,
-                           column: 3
-                         }
-                       ],
-                       path: [
-                         "plan"
-                       ],
-                       extensions: %{
-                         classification: "DataFetchingException"
-                       }
-                     }
-                   ]
-                 })
-               end)
+      {{:error, [error]}, log} =
+        with_log(fn ->
+          validate_body(%{
+            data: %{plan: nil},
+            errors: [
+              %{
+                message:
+                  "Exception while fetching data (/plan) : The value is not in range[0.0, 1.7976931348623157E308]: -5.0",
+                locations: [
+                  %{
+                    line: 2,
+                    column: 3
+                  }
+                ],
+                path: [
+                  "plan"
+                ],
+                extensions: %{
+                  classification: "DataFetchingException"
+                }
+              }
+            ]
+          })
+        end)
+
+      assert error == %OpenTripPlannerClient.Error{
+               details: %{
+                 path: ["plan"],
+                 extensions: %{classification: "DataFetchingException"},
+                 locations: [%{line: 2, column: 3}]
+               },
+               message:
+                 "Exception while fetching data (/plan) : The value is not in range[0.0, 1.7976931348623157E308]: -5.0",
+               type: :graphql_error
+             }
 
       assert log =~ "Exception while fetching data"
     end
 
     test "handles routing errors" do
-      assert {{:error, "PATH_NOT_FOUND"}, log} =
+      assert {{:error, error}, log} =
                with_log(fn ->
                  validate_body(%{
                    data: %{plan: %{routing_errors: [%{code: "PATH_NOT_FOUND"}]}}
                  })
                end)
 
+      assert error == [
```

Comment:
Nit (non-blocking): `error` in this test is a list of errors. Could you either call it `errors` or destructure it as `[error]` on line 105 where it's initialized?